### PR TITLE
Fixed #34488 -- Made ClearableFileInput preserve "Clear" checked attribute when form is invalid.

### PR DESCRIPTION
**Pull request by mgaligniana**
Hello!

It fixes [34488](https://code.djangoproject.com/ticket/34488)

To run tests please go to `tests/` folder and run `./runtests.py admin_widgets.tests.ImageFieldWidgetsSeleniumTests --selenium=<browser> --parallel=1` with "browser" as `firefox`, `chrome` or `safari`.

Thank you!

### CODE REVIEW

File: tests/admin_widgets/tests.py [hunk 1]
```diff
@@ -1772,39 +1772,60 @@ def test_ForeignKey_using_to_field(self):
 
 @skipUnless(Image, "Pillow not installed")
 class ImageFieldWidgetsSeleniumTests(AdminWidgetSeleniumTestCase):
-    def test_clearablefileinput_widget(self):
+    NAME_INPUT_ID = "id_name"
+    PHOTO_INPUT_ID = "id_photo"
+    SAVE_AND_EDIT_BUTTON_CSS_SELECTOR = "input[value='Save and continue editing']"
+    TESTS_FILES_FOLDER = f"{os.getcwd()}/files"
```
Comment (felixxm) [inline tests/admin_widgets/tests.py:1778]: We don't allow calling functions in f-strings.

File: tests/admin_widgets/tests.py [hunk 2]
```diff
@@ -1772,39 +1772,60 @@ def test_ForeignKey_using_to_field(self):
 
 @skipUnless(Image, "Pillow not installed")
 class ImageFieldWidgetsSeleniumTests(AdminWidgetSeleniumTestCase):
-    def test_clearablefileinput_widget(self):
+    NAME_INPUT_ID = "id_name"
+    PHOTO_INPUT_ID = "id_photo"
+    SAVE_AND_EDIT_BUTTON_CSS_SELECTOR = "input[value='Save and continue editing']"
```
Comment (felixxm) [inline tests/admin_widgets/tests.py:1777]: `SAVE_AND_EDIT_BUTTON_CSS_SELECTOR` is used in a single place, so this attribute is unnecessary.

File: tests/admin_widgets/tests.py [hunk 3]
```diff
@@ -1813,12 +1834,28 @@ def _submit_and_wait():
...
+        self.selenium.find_element(By.ID, self.CLEAR_CHECKBOX_ID).click()
+        self._submit_and_wait()
+        self.assertEqual(
+            self.selenium.find_element(By.CSS_SELECTOR, ".errorlist li").text,
+            "This field is required.",
+        )
+        # "Clear" persists checked.
+        self.assertTrue(
+            self.selenium.find_element(By.ID, self.CLEAR_CHECKBOX_ID).is_selected()
+        )
```
Comment (felixxm) [inline tests/admin_widgets/tests.py:1861]: Some of existing tests in `ClearableFileInputTest` already fail, so maybe we don't need new selenium tests :thinking:

File: tests/admin_widgets/tests.py (hunk 1 above)
Comment (mgaligniana) [reply tests/admin_widgets/tests.py:1778]: Thank you! I didn't know it and I just found it in https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style

File: tests/admin_widgets/tests.py (hunk 3 above)
Comment (mgaligniana) [reply tests/admin_widgets/tests.py:1860]: When I was debugging it (from the browser) and I edited the `checked` attribute manually sometimes it didn't take the change 🤔 , so running from the browser gives me more confidence than just assert the html during the html render.

**Comment by mgaligniana** [comment]
buildbot, test on selenium.

File: django/contrib/admin/templates/admin/widgets/clearable_file_input.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {% if widget.is_initial %}<p class="file-upload">{{ widget.initial_text }}: <a href="{{ widget.value.url }}">{{ widget.value }}</a>{% if not widget.required %}
 <span class="clearable-file-input">
-<input type="checkbox" name="{{ widget.checkbox_name }}" id="{{ widget.checkbox_id }}"{% if widget.attrs.disabled %} disabled{% endif %}>
+<input type="checkbox" name="{{ widget.checkbox_name }}" id="{{ widget.checkbox_id }}"{% if widget.attrs.disabled %} disabled{% endif %}{% if widget.attrs.checked %} checked{% endif %}>
```
Comment (felixxm) [inline django/contrib/admin/templates/admin/widgets/clearable_file_input.html:3]: Can we use `{% include "django/forms/widgets/attrs.html" %}` instead?
```suggestion
<input type="checkbox" name="{{ widget.checkbox_name }}" id="{{ widget.checkbox_id }}"{% include "django/forms/widgets/attrs.html" %}>
```

Comment (mgaligniana) [reply django/contrib/admin/templates/admin/widgets/clearable_file_input.html:3]: Yes!! Thank you!!

**Review by felixxm — COMMENTED** [review]
@mgaligniana Thanks :+1:

**Review by felixxm — APPROVED** [review]

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**